adapters/sharethrough/sharethrough_test.go

```diff
@@ -1,278 +1,24 @@
-					},
-				}},
-			},
-			expected: []*adapters.RequestData{stubReq},
-		},
-	}
-
-	mockAdServer := MockStrAdServer{
-		mockRequestFromOpenRTB: func() (*adapters.RequestData, error) {
-			return stubReq, nil
-		},
-	}
-
-	adapter := SharethroughAdapter{AdServer: mockAdServer}
-	for testName, test := range tests {
-		t.Logf("Test case: %s\n", testName)
-
-		output, actualErrors := adapter.MakeRequests(test.input, &adapters.ExtraRequestInfo{})
-
-		if len(output) != 1 {
-			t.Errorf("Expected one request in result, got %d\n", len(output))
-			return
-		}
-
-		assertRequestDataEquals(t, testName, test.expected[0], output[0])
-		if len(actualErrors) != 0 {
-			t.Errorf("Expected no errors, got %d\n", len(actualErrors))
-		}
-	}
-}
-
-func TestFailureMakeRequests(t *testing.T) {
-	tests := map[string]struct {
-		input    *openrtb2.BidRequest
-		expected string
-	}{
-		"Returns nil if failed to generate request": {
-			input: &openrtb2.BidRequest{
-				Site: &openrtb2.Site{
-					Page: "test.com",
-				},
-				Device: &openrtb2.Device{
-					UA: "Android Chome/60",
-				},
-				Imp: []openrtb2.Imp{{
-					ID:  "abc",
-					Ext: []byte(`{"pkey": "pkey", "iframe": true, "iframeSize": [10, 20]}`),
-					Banner: &openrtb2.Banner{
-						Format: []openrtb2.Format{{H: 30, W: 40}},
-					},
-				}},
-			},
-			expected: "error generating request",
-		},
-	}
-
-	mockAdServer := MockStrAdServer{
-		mockRequestFromOpenRTB: func() (*adapters.RequestData, error) {
-			return nil, fmt.Errorf("error generating request")
-		},
-	}
-
-	adapter := SharethroughAdapter{AdServer: mockAdServer}
-	for testName, test := range tests {
-		t.Logf("Test case: %s\n", testName)
-
-		output, actualErrors := adapter.MakeRequests(test.input, &adapters.ExtraRequestInfo{})
-
-		if output != nil {
-			t.Errorf("Expected result to be nil, got %d elements\n", len(output))
-		}
-		if len(actualErrors) != 1 {
-			t.Errorf("Expected one error, got %d\n", len(actualErrors))
-		}
-		if actualErrors[0].Error() != test.expected {
-			t.Errorf("Error mismatch: expected '%s' got '%s'\n", test.expected, actualErrors[0].Error())
-		}
-	}
-}
-
-func TestSuccessMakeBids(t *testing.T) {
-	stubBidderResponse := adapters.BidderResponse{}
-
-	tests := map[string]struct {
-		inputResponse *adapters.ResponseData
-		expected      *adapters.BidderResponse
-	}{
-		"Returns nil,nil if ad server responded with no content": {
-			inputResponse: &adapters.ResponseData{
-				StatusCode: http.StatusNoContent,
-			},
-			expected: nil,
-		},
-		"Generates response if ad server responded with 200": {
-			inputResponse: &adapters.ResponseData{
-				StatusCode: http.StatusOK,
-				Body:       []byte(`{}`),
-			},
-			expected: &stubBidderResponse,
-		},
-	}
-
-	mockAdServer := MockStrAdServer{
-		mockResponseToOpenRTB: func() (*adapters.BidderResponse, []error) {
-			return &stubBidderResponse, []error{}
-		},
-	}
+func TestJsonSamples(t *testing.T) {
+	version.Ver = "42.42"
```

Comment:
This is a bit scary since you're changing the shared definition for all tests running after this. I recommend you either write your tests against the default empty string or capture the version's state in your adapter struct and modify it there. I don't care as much about the state of your adapterVersion as it's already isolated to just this test.